packages/editor/src/lib/editor/Editor.ts

```diff
@@ -7714,19 +7716,20 @@ export class Editor extends EventEmitter<TLEventMap> {
 	 *
 	 * @public
 	 */
-	setTemporaryAssetPreview(assetId: TLAssetId, file: File) {
-		this.temporaryAssetPreview[assetId] = URL.createObjectURL(file)
-	}
+	createTemporaryAssetPreview(assetId: TLAssetId, file: File) {
+		assert(!this.temporaryAssetPreview.has(assetId), 'Asset preview already exists')
 
-	/**
-	 * Clean up memory after using registerTemporaryAssetPreview.
-	 *
-	 * @param objectUrl - The url to revoke.
-	 *
-	 * @public
-	 */
-	clearTemporaryAssetPreview(objectUrl: string) {
-		URL.revokeObjectURL(objectUrl)
+		const urlString = URL.createObjectURL(file)
+		const urlResource = new RC(
+			urlString,
+			() => {
+				this.temporaryAssetPreview.delete(assetId)
+				URL.revokeObjectURL(urlString)
+			},
+			60_000
```

Comment:
nit: we should write as 60 * 1000 - i dunno, i think it's typically not written as the complete number for readability (in our codebase either)